include/sys/sys_heap.h

```diff
@@ -61,6 +61,37 @@ struct z_heap_stress_result {
 	uint64_t accumulated_in_use_bytes;
 };
 
+#ifdef CONFIG_SYS_HEAP_RUNTIME_STATS
+
+struct sys_heap_runtime_stats {
+	size_t free_bytes;
+	size_t allocated_bytes;
+	/* The max available bytes at one time */
+	size_t max_avail_bytes;
+};
+
+/**
+ * @brief Get the runtime statistics of a sys_heap
+ *
+ * @param heap Pointer to specified sys_heap
+ * @param stats_t Pointer to struct to copy statistics into
+ * @return -EINVAL if null pointers, otherwise 0
+ */
+int heap_runtime_stats_get(struct sys_heap *heap,
```

Comment:
Naming: should be "sys_heap_*()".  sys_ is the prefix (one of them) for a public API.